docs/user-guide/node-selection/node-selection.md

```diff
@@ -0,0 +1,200 @@
+zone XYZ, but if it's not possible, then allow some to run elsewhere".
+
+Node affinity is currently expressed using an annotation on Pod. Once the feature goes to GA it will use a field of Pod.
+
+Here's an example of a pod that uses node affinity:
+
+{% include code.html language="yaml" file="pod-with-node-affinity.yaml" ghlink="/docs/user-guide/node-selection/pod-with-node-affinity.yaml" %}
+
+This node affinity rule says the pod can only be placed on a node with a label whose key is
+`kubernetes.io/e2e-az-name` and whose value is either `e2e-az1` or `e2e-az2`. In addition,
+among nodes that meet that criteria, nodes with a label whose key is `another-annotation-key` and whose
+value is `another-annotation-value` should be preferred.
+
+You can see the operator `In` being used in the example. The new node affinity syntax supports the following operators: `In`, `NotIn`, `Exists`, `DoesNotExist`, `Gt`, `Lt`.
+There is no explicit "node anti-affinity" concept, but `NotIn` and `DoesNotExist` give that behavior.
+
+If you specify both `nodeSelector` and `nodeAffinity`, *both* must be satisfied for the pod
+to be scheduled onto a candidate node.
+
+For more information on node affinity, see the design doc
+[here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/nodeaffinity.md).
+
+### Inter-pod affinity and anti-affinity (alpha feature)
+
+Inter-pod affinity and anti-affinity were introduced in Kubernetes 1.4.
+Inter-pod affinity and anti-affinity allow you to constrain which nodes your pod is eligible to schedule on *based on
+labels on pods that are already running on the node* rather than based on labels on nodes. The rules are of the form "this pod should (or, in the case of
+anti-affinity, should not) run in an X if that X is already running one or more pods that meet rule Y." Y is expressed
+as a LabelSelector with an associated list of namespaces (or "all" namespaces); unlike nodes, because pods are namespaced
+(and therefore the labels on pods are implicitly namespaced),
+a label selector over pod labels must specify which namespaces the selector should apply to. Conceptually X is a topology domain
+like node, rack, cloud provider zone, cloud provider region, etc. You express it using a `topologyKey` which is the
+key for the node label that the system uses to denote such a topology domain, e.g. see the label keys listed above
+in the section "Interlude: built-in node labels."
+
+As with node affinity, there are currently two types of pod affinity and anti-affinity, called `requiredDuringSchedulingIgnoredDuringExecution` and
+`preferredDuringSchedulingIgnoredDuringExecution` which denote "hard" vs. "soft" requirements.
+See the description in the node affinity section earlier.
+An example of `requiredDuringSchedulingIgnoredDuringExecution` affinity would be "co-locate the pods of service A and service B
```

Comment:
Consider rewording and split into two sentences. These examples are also repeated in another doc that can probably be linked.